### DISCUSSION TITLE
Form request validated data is not casted to enum

### DISCUSSION THREAD

**Discussion by nicolasbeauvais** [spatie/laravel-enum#114] [Ideas]
Hello :wave:,

When casting enums in a Form Request, only the data payload in the Request instance is getting cast. So if following best practice and getting the data with the `validated` method in a controller, the original pre-cast data will be returned.

```php
class Controller
{
    public function index(MyFormRequest $request)
    {  
        $request->validated()); // Doesn't return cast data

        $request->get('status'); // Return cast data
    }
}
```

As a fix, I created an enhanced `TransformsEnums` trait implementing the following method:

```php
public function withValidator(Validator $validator)
{
    $validator->after(function (Validator $validator) {
        $data = $validator->getData();

        /** @var Enum $enumClass */
        foreach ($this->enums() as $key => $enumClass) {
            Arr::set($data, $key, $enumClass::from($this[$key]));
        }

        $validator->setData($data);
    });
}
```

Do you think that this is something that could be added to the package?

I could draft a PR on this, but given the native enum support on PHP 8.1 I wanted to make sure you would continue the maintenance of laravel-enum before working on it.

Cheers.

[a comment by an automated account is not shown]

**Comment by nicolasbeauvais** [comment]
Not stale

**Comment by Jasperscheper** [comment]
Any updates on this?
